app/presenters/ht_registration_presenter.rb

```diff
@@ -79,11 +83,17 @@ def show_detail_display_name
   end
 
   def show_detail_edu_person_principal_name
-    env["HTTP_X_SHIB_EDUPERSONPRINCIPALNAME"]
+    name = env["HTTP_X_SHIB_EDUPERSONPRINCIPALNAME"] || ""
+    # Check if principal name or email matches the given DSP email
+    badge = name == applicant_email ? :ok : :mismatch
+    name + " " + BADGES[badge].label_span
   end
 
   def show_detail_email
-    env["HTTP_X_SHIB_MAIL"]
+    email = env["HTTP_X_SHIB_MAIL"] || ""
+    # Check if principal name or email matches the given DSP email
+    badge = email == applicant_email ? :ok : :mismatch
```

Comment:
As with eppn - it's not "danger do not approve" if they don't match, but should invite extra scrutiny; likewise not all IdPs will send it, so not really a problem if it's missing entirely